osf/models/user.py

```diff
@@ -777,6 +779,24 @@ def merge_user(self, user):
 
         user.save()
 
+    def _merge_users_preprints(self, user):
+        from osf.models.preprint import Preprint, PreprintContributor
+
+        preprints = Preprint.objects.filter(creator=user)
+        for preprint in preprints:
+            preprint.creator = self
+            preprint.save()
+
+        preprint_contributors = PreprintContributor.objects.filter(user=user)
+
+        # If both users are already contribs we will violate unique constraint so just remove old contrib.
+        for preprint_contributor in preprint_contributors:
+            if self in preprint_contributor.preprint.contributors.all():
+                preprint_contributor.delete()
+            else:
+                preprint_contributor.user = self
+                preprint_contributor.save()
```

Comment:
Preprint permissions are not stored in the PreprintContributor table - this table is only for order and bibliographic information.  We need to also manually add permissions to `self` and remove permissions from `user` - these perms are stored in guardian tables.


(as a side note, functions like Node.add_contributor take care of both creating the PreprintContributor object, storing bibligraphic and order information, and then add the appropriate permissions which are stored in the guardian table.  I wouldn't use that method here though, since it also adds logs. Instead, just manually modify the PreprintContributor object that you're already doing and then update permissions).

for this branch, then, something like:
```python
...
else:
  preprint_contributor.user = self
  user_perm = preprint_contributor.permission
  preprint_contributor.preprint.add_permission(self, user_perm)
  preprint_contributor.preprint.remove_permission(user, user_perm)
  preprint_contributor.save()
```